### PR TITLE
ADEN-3840 Add floorAdhesion

### PR DESCRIPTION
**Pull request by v3rth**
## Links

* https://wikia-inc.atlassian.net/browse/ADEN-3840

### CODE REVIEW

**Comment by bognix** [comment]
fix unit tests

File: front/main/app/routes/wiki-page.js [hunk 1]
```diff
@@ -213,6 +214,12 @@ export default Ember.Route.extend(
 			 * @returns {boolean}
 			 */
 			didTransition() {
+				let adsHighImpact = this.get('adsHighImpact');
+
+				$script.ready(M.prop('adsUrl'), () => {
```
Comment (bognix) [inline front/main/app/routes/wiki-page.js]: can you move the logic with waiting for $script.ready to adsHighImpact service

File: front/main/app/services/ads-high-impact.js [hunk 2]
```diff
@@ -3,9 +3,22 @@ import Ads from 'common/modules/ads';
 
 export default Ember.Service.extend({
 	name: 'INVISIBLE_HIGH_IMPACT_2',
-
+	ads: Ads.getInstance(),
 	reload() {
-		const ads = Ads.getInstance();
-		ads.pushSlotToQueue(this.get('name'));
+		this.ads.pushSlotToQueue(this.get('name'));
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: could you move the logic for deciding if interstitial should be loaded from `routes/application` to `reload` method

File: front/main/app/styles/module/_ads.scss [hunk 3]
```diff
@@ -83,6 +83,35 @@
 	border: 0;
 }
 
+#InvisibleHighImpactWrapper.floor-adhesion {
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: remove background from top-bar (and possibly height)

File: front/main/app/styles/module/_ads.scss [hunk 4]
```diff
@@ -83,6 +83,35 @@
...
+	bottom: 0;
+	position: fixed;
+
+	.background, div.label {
+		display: none;
+	}
+
+	a.close {
+		background-color: #000;
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: use variable instead of color

File: front/main/app/styles/module/_ads.scss [hunk 5]
```diff
@@ -83,6 +83,35 @@
...
+		border-radius: 100%;
+		border: 2px solid white;
+		float: right;
+		height: 40px;
+		padding: 5px;
+		position: relative;
+		text-align: center;
+		top: 20px;
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: top should be `-20px`

File: front/main/app/styles/module/_ads.scss [hunk 6]
```diff
@@ -83,6 +83,35 @@
...
+		width: 40px;
+		z-index: 1;
+
+		svg {
+			display: inline-block;
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: make sure this styles are needed

File: front/main/app/styles/module/_ads.scss [hunk 7]
```diff
@@ -83,6 +83,35 @@
...
+		border-radius: 100%;
+		border: 2px solid white;
+		float: right;
+		height: 40px;
+		padding: 5px;
+		position: relative;
+		text-align: center;
+		top: 20px;
+		width: 40px;
+		z-index: 1;
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: I believe this one is not needed

File: front/main/app/styles/module/_ads.scss (hunk 4 above)
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: part of this styles are duplicated with https://github.com/Wikia/mercury/pull/2749/files#diff-2d9e3814826da5e868b797cdfd6b6005R158. Please clean up

File: front/main/app/styles/module/_ads.scss (hunk 7 above)
Comment (v3rth) [inline front/main/app/styles/module/_ads.scss]: nope :(

File: front/main/app/styles/module/_ads.scss (hunk 3 above)
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: @v3rth  I still see the top bar on your branch

Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: and style by class not by id please

File: front/main/app/styles/module/_ads.scss [hunk 8]
```diff
@@ -83,6 +83,37 @@
...
+		float: right;
+		height: $closeButtonHeight;
+		padding: 5px;
+		position: relative;
+		text-align: center;
+		top: $closeButtonHeight / 2;
+		width: $closeButtonHeight;
+		z-index: 1;
+
+		svg {
```
Comment (bognix) [inline front/main/app/styles/module/_ads.scss]: this part is not needed.

File: front/main/app/services/ads-high-impact.js [hunk 9]
```diff
@@ -3,9 +3,41 @@ import Ads from 'common/modules/ads';
...
+
+	isProperGeo(param) {
+		return Ember.get(Wikia, 'geo.isProperGeo')(param);
+	},
+
+	isInvisibleHighImpactEnabled() {
+		return this.isProperGeo && this.isProperGeo(this.highImpactCountries);
+	},
 
 	reload() {
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: rename method so it's consistent with `loadFloorAdhesion`

File: front/main/app/services/ads-high-impact.js [hunk 10]
```diff
@@ -3,9 +3,41 @@ import Ads from 'common/modules/ads';
...
+	},
+
+	loadFloorAdhesion() {
+		if (this.isInvisibleHighImpactEnabled() && this.isProperGeo(this.floorAdhesionCountries)) {
+			this.ads.addSlot(this.get('name'));
+		}
+	},
+
+	loadFloorAdhesionWhenPossible() {
+		let self = this;
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: use `const` not `let` if you are not modifying the variable.

File: front/main/app/services/ads-high-impact.js [hunk 11]
```diff
@@ -3,9 +3,41 @@ import Ads from 'common/modules/ads';
...
+		if (M.prop('adsUrl')) {
+			$script.ready(M.prop('adsUrl'), () => {
+				self.loadFloorAdhesion();
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: you don't need to use `self` trick, arrow functions are here to solve this kind of problems. Simply use `this` inside the method.

File: front/main/app/services/ads-high-impact.js [hunk 12]
```diff
@@ -3,9 +3,41 @@ import Ads from 'common/modules/ads';
 
 export default Ember.Service.extend({
 	name: 'INVISIBLE_HIGH_IMPACT_2',
+	ads: Ads.getInstance(),
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: there is a fancy service called `ads-state` inside this service there is a property `module` which returns you what you need here - use this service.

File: front/main/app/services/ads-high-impact.js [hunk 13]
```diff
@@ -1,11 +1,39 @@
...
+	interstitialOnTransitionCountries:
+		Ember.get(Wikia, 'InstantGlobals.wgAdDriverMobileTransitionInterstitialCountries'),
+	floorAdhesionCountries:
+		Ember.get(Wikia, 'InstantGlobals.wgAdDriverMobileFloorAdhesionCountries'),
 
-	reload() {
-		const ads = Ads.getInstance();
-		ads.pushSlotToQueue(this.get('name'));
+	isProperGeo(param) {
+		const fun = Ember.get(Wikia, 'geo.isProperGeo');
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: please don't use the name `fun`

**Comment by bognix** [comment]
LGTM

**Pull request merged**